### PR TITLE
refactor: rewrite the K8s content

### PR DESCRIPTION
**Pull request by zyy17**
## What's Changed in this PR

Almost rewrite the Kubernetes section base on more details:

1. Add the step by step getting start to create the greptimedb cluster;
2. Refactor the documents of managing greptimedb operator;

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+You can use the following command to search the latest version of the GreptimeDB Operator:
+
+```bash
+helm search repo greptime/greptimedb-operator
+```
+
+:::note
+The output may vary depending on the version of the charts.
+:::
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:74]:
```suggestion
The output may vary depending on the version of the charts.
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+</details>
+
+:::tip
+If you want to list all the available versions, you can use the following command:
+
+```
+helm search repo greptime/greptimedb-operator --versions
+```
+:::
+
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:92]: please do not use tip for the listing versions, move it before `Upgrade the GreptimeDB Operator` to check all the available versions.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+You can upgrade to the latest released version of the GreptimeDB Operator by running the following command:
+
+```bash
+helm --namespace greptimedb-admin upgrade greptimedb-operator greptime/greptimedb-operator
+```
+
+:::note
+The output may vary depending on the version of the charts.
+:::
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:103]:
```suggestion
The output may vary depending on the version of the charts.
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+```
+
+:::tip
+You can use one command to install or upgrade the GreptimeDB Operator with the custom configuration:
+
+```bash
+helm --namespace greptimedb-admin upgrade --install \
+  greptimedb-operator greptime/greptimedb-operator -f values.yaml
+```
+:::
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:218]: please do not use `tip`, this is a feature that should be introduced in the main text.

**Review by nicecui — COMMENTED** [review]
Please reduce the use of NOTE and TIP tags, as a large number of color blocks will affect the user's reading experience.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md (hunk 4 above)
Comment (zyy17) [reply docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md]: It's not a feature. It's like some kind of `helm` tip.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
+# GreptimeDB Operator Management
+
+## Prerequisites
+
+- Kubernetes >= v1.18.0
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) >= v1.18.0
+- [Helm](https://helm.sh/docs/intro/install/) >= v3.0.0
+
+## Quick Start
```
[the remark this answers is not in the document]
Comment (zyy17) [reply docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:39]: It's more like some special quick start for greptimedb-operator. It's useful for some testing environments if just want to install operator.

File: docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+  <summary>Expected Output</summary>
+```bash
+NAME                          	CHART VERSION	APP VERSION  	DESCRIPTION
+greptime/greptimedb-cluster   	0.2.25       	0.9.5        	A Helm chart for deploying GreptimeDB cluster i...
+greptime/greptimedb-operator  	0.2.9        	0.1.3-alpha.1	The greptimedb-operator Helm chart for Kubernetes.
+greptime/greptimedb-standalone	0.1.27       	0.9.5        	A Helm chart for deploying standalone greptimedb
+```
+</details>
+
+## Install the GreptimeDB Operator
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md:90]:
```suggestion
## Install and verify the GreptimeDB Operator
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+## Production Deployment
+
+For production deployments, it's recommended to use Helm to install the GreptimeDB Operator.
+
+### Installation
+
+You can refer [Install the GreptimeDB Operator](/user-guide/deployments/deploy-on-kubernetes/getting-started.md#install-the-greptimedb-operator) for detailed instructions.
+
+:::note
+If you are using [Argo CD](https://argo-cd.readthedocs.io/en/stable/) to deploy applications, please make sure that the `Application` has set the [`ServerSideApply=true`](https://argo-cd.readthedocs.io/en/latest/user-guide/sync-options/#server-side-apply) to enable the server-side apply(other GitOps tools may have similar settings).
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:48]:
```suggestion
If you are using [Argo CD](https://argo-cd.readthedocs.io/en/stable/) to deploy applications, please make sure that the `Application` has set the [`ServerSideApply=true`](https://argo-cd.readthedocs.io/en/latest/user-guide/sync-options/#server-side-apply) to enable the server-side apply (other GitOps tools may have similar settings).
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+<details>
+<summary>Expected Output</summary>
+```bash
+NAME                        	CHART VERSION	APP VERSION  	DESCRIPTION
+greptime/greptimedb-operator	0.2.9        	0.1.3-alpha.1	The greptimedb-operator Helm chart for Kubernetes.
+```
+</details>
+
+#### Upgrade the GreptimeDB Operator
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:86]: maybe we can move this title before line 71. When users find out there is a newer version, then can choose to update the Operator.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+```
+
+<details>
+<summary>Expected Output</summary>
+```bash
+NAME                        	CHART VERSION	APP VERSION  	DESCRIPTION
+greptime/greptimedb-operator	0.2.9        	0.1.3-alpha.1	The greptimedb-operator Helm chart for Kubernetes.
+```
+</details>
+
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:85]: or list all the available versions:

```bash
helm search repo greptime/greptimedb-operator --versions
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+```bash
+helm -n greptimedb-admin upgrade greptimedb-operator greptime/greptimedb-operator --version <version>
+```
+
+You can use the following command to list all the available versions:
+
+```bash
+helm search repo greptime/greptimedb-operator --versions
+```
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:128]: please move these lines to line 85

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+<summary>Expected Output</summary>
+```bash
+NAME               	NAMESPACE	REVISION	UPDATED                             	STATUS  	CHART                    	APP VERSION
+greptimedb-operator	default  	2       	2024-10-28 19:30:52.62097 +0800 CST 	deployed	greptimedb-operator-0.2.9	0.1.3-alpha.1
+```
+</details>
+
+### CRDs
+
+There are two kind of CRD that are installed with the GreptimeDB Operator: `GreptimeDBCluster` and `GreptimeDBStandalone`.
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:146]:
```suggestion
There are two kinds of CRD that are installed with the GreptimeDB Operator: `GreptimeDBCluster` and `GreptimeDBStandalone`.
```

File: docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md [hunk 12]
```diff
@@ -0,0 +1,442 @@
...
+```
+</details>
+
+## Install the GreptimeDB cluster with monitoring integration
+
+Now that the GreptimeDB Operator and etcd cluster are installed, you can deploy a minimum GreptimeDB cluster with monitoring integration:
+
+:::warning
+The default configuration for the GreptimeDB cluster is not recommended for production use. 
+You should adjust the configuration according to your requirements.
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md:265]:
```suggestion
You should [adjust the configuration](greptimedb-operator-management.md#configuration) according to your requirements.
```

File: docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,442 @@
...
+
+- [Docker](https://docs.docker.com/get-started/get-docker/) >= v23.0.0
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) >= v1.18.0
+- [Helm](https://helm.sh/docs/intro/install/) >= v3.0.0
+- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/) >= v0.20.0
+
+## Create a test Kubernetes cluster
+
+:::warning
+Don't use the `kind` for production use. It's recommended to use a managed Kubernetes service like [Amazon EKS](https://aws.amazon.com/eks/), [Google GKE](https://cloud.google.com/kubernetes-engine/), or [Azure AKS](https://azure.microsoft.com/en-us/services/kubernetes-service/).
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md:19]: please add AKS to `typos.toml`

File: docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md (hunk 12 above)
Comment (zyy17) [reply docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md:265]: It's the cluster configuration. The docs are still wip.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+```
+</details>
+
+By default, the GreptimeDB Operator chart will manage the installation and upgrade of the CRDs and the users don't need to manage them manually.
+
+### Configuration
+
+The GreptimeDB Operator chart provides a set of configuration options that allow you to customize the installation, you can refer to the [GreptimeDB Operator Helm Chart](https://github.com/GreptimeTeam/helm-charts/blob/main/charts/greptimedb-operator/README.md) for more details.
+
+You can create a `values.yaml` to configure the GreptimeDB Operator chart, for example:
```
[the remark this answers is not in the document]
Comment (zyy17) [reply docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:168]: the `values.yaml` in helm-charts contains all the configuration.

File: docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md [hunk 15]
```diff
@@ -0,0 +1,429 @@
...
+Forwarding from 127.0.0.1:4002 -> 4002
+Forwarding from [::1]:4002 -> 4002
+Forwarding from 127.0.0.1:4003 -> 4003
+Forwarding from [::1]:4003 -> 4003
+```
+</details>
+
+:::note
+If you want to expose the service to the public, you can use the kubectl port-forward command with the `--address` option:
+
```
[the remark this answers is not in the document]
Comment (zyy17) [reply docs/user-guide/deployments/deploy-on-kubernetes/getting-started.md:369]: I change to `warning`.

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+helm -n greptimedb-admin install greptimedb-operator greptime/greptimedb-operator -f values.yaml
+```
+
+If you want to upgrade the GreptimeDB Operator with the custom configuration, you can use the following command:
+
+```bash
+helm -n greptimedb-admin upgrade greptimedb-operator greptime/greptimedb-operator -f values.yaml
+```
+
+:::tip
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:196]:
```suggestion
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+You can use one command to install or upgrade the GreptimeDB Operator with the custom configuration:
+
+```bash
+helm -n greptimedb-admin upgrade --install \
+  greptimedb-operator greptime/greptimedb-operator -f values.yaml
+```
+:::
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:203]:
```suggestion
```

File: docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (nicecui) [reply docs/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:158]: can you give a link to the `values.yaml` file?

**Review by nicecui — COMMENTED** [review]
This PR contains a lot of work, thank you for your contribution.
The English documents LGTM.

**Review by nicecui — APPROVED** [review]

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+### 配置
+
+GreptimeDB Operator chart 提供了一组配置选项，允许您自行配置，您可以参考 [GreptimeDB Operator Helm Chart](https://github.com/GreptimeTeam/helm-charts/blob/main/charts/greptimedb-operator/README.md) 来获取更多详细信息。
+
+你可以创建一个 `values.yaml` 来配置 GreptimeDB Operator chart，例如：
+
+```yaml
+image:
+  # -- The image registry
+  registry: docker.io
```
[the remark this answers is not in the document]
Comment (zyy17) [reply i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/greptimedb-operator-management.md:163]: I add the note.

**Review by daviderli614 — APPROVED** [review]

**Pull request merged**